core/src/main/java/org/geysermc/geyser/skin/FakeHeadProvider.java

```diff
@@ -91,11 +94,38 @@ public SkinProvider.SkinData load(@Nonnull FakeHeadEntry fakeHeadEntry) throws E
                 }
             });
 
-    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag profileTag) {
-        SkinManager.GameProfileData gameProfileData = SkinManager.GameProfileData.from(profileTag);
-        if (gameProfileData == null) {
-            return;
+    public static void setHead(GeyserSession session, PlayerEntity entity, CompoundTag NBTData) {
+        if (NBTData.get("SkullOwner") instanceof CompoundTag profileTag) {
+            SkinManager.GameProfileData gameProfileData = SkinManager.GameProfileData.from(profileTag);
+            if (gameProfileData == null) {
+                return;
+            }
+            loadHead(session, entity, gameProfileData);
+        } else if (NBTData.get("SkullOwner") instanceof StringTag ownerTag) {
+            String owner = ownerTag.getValue();
+            if (owner == null) {
```

Comment:
Not sure if this can ever *be* null. But we can do a blank/empty check for sure.